src/mlpack/methods/ann/layer/noisylinear_impl.hpp

```diff
@@ -48,6 +48,50 @@ NoisyLinear<InputDataType, OutputDataType>::NoisyLinear(
   biasEpsilon.set_size(outSize, 1);
 }
 
+template<typename InputDataType, typename OutputDataType>
+NoisyLinear<InputDataType, OutputDataType>::NoisyLinear(
+    NoisyLinear&& layer) :
+    inSize(std::move(layer.inSize)),
+    outSize(std::move(layer.outSize)),
+    weights(std::move(layer.weights))
+{
+  layer.inSize = 0;
+  layer.outSize = 0;
+  layer.weights = nullptr;
+  Reset();
+}
+
+template<typename InputDataType, typename OutputDataType>
+NoisyLinear<InputDataType, OutputDataType>&
+NoisyLinear<InputDataType, OutputDataType>::operator=(NoisyLinear const& layer)
+{
+  if( this != &layer)
```

Comment:
```suggestion
  if (this != &layer)
```
Another simple style fix.